### DISCUSSION TITLE
How did you get AMD to work with Comfy on Windows?

### DISCUSSION THREAD

**Discussion by BrechtCorbeel** [comfyanonymous/ComfyUI#6195] [General]
I get a lot of circling around to Linux, but as that failed for me I am not too shabby going Linux with AMD, AMD install seems fair on Windows, but it seems getting it to build with ROCm or HIP as I understand it is not ready.

Been trying docker but am failing.

**Comment by ltdrdata** [comment]
AFAIK, there is a way to use DirectML on Windows.

https://github.com/comfyanonymous/ComfyUI#directml-amd-cards-on-windows

**Reply by BrechtCorbeel** [reply]
The issue I have not gotten this to work either and from what I read directml, ontop of AMD being slow is slower then rocM.

At this point if the RX7900XTX I have is slower I might have just bought a 4060 it is that slow.

The 3D rendering performance of the RX7900 cannot compete with my dual GPU machine with a GTX1070+RTX3060 combo. Which are 2 older lower end cards against AMD's flagship GPU.

AMD is sad in the GPU market and Nvidia is becoming a cash grab company and failing in its own market too because of it.

**Comment by VeteranXT** [comment]
https://github.com/CS1o/Stable-Diffusion-Info/wiki/Webui-Installation-Guides Follow this guide. It enables zluda on Windows.

**Comment by zankarl** [comment]
I managed to run it on the AMD GPU, but DirectML does not have memory control. I can use it with small models like the Hidream and with images equivalent to 512x512. If it exceeds this size, the latent space exceeds the size of the VRAM. I am doing tests to control the memory through the Vulkan API. I will post any news here.
